src/analyses/variable-sensitivity/variable_sensitivity_object_factory.cpp

```diff
@@ -10,8 +10,217 @@
 #include "util/namespace.h"
 #include "value_set_abstract_value.h"
 
-variable_sensitivity_object_factoryt::ABSTRACT_OBJECT_TYPET
-variable_sensitivity_object_factoryt::get_abstract_object_type(const typet type)
+vsd_configt vsd_configt::from_options(const optionst &options)
+{
+  vsd_configt config{};
+
+  if(
+    options.get_bool_option("value-set") &&
+    options.get_bool_option("data-dependencies"))
+  {
+    throw invalid_command_line_argument_exceptiont{
+      "Value set is not currently supported with data dependency analysis",
+      "--value-set --data-dependencies",
+      "--data-dependencies"};
+  }
+
+  config.value_abstract_type =
+    option_to_abstract_type(options, "values", value_option_mappings, CONSTANT);
+
+  config.pointer_abstract_type = option_to_abstract_type(
+    options, "pointers", pointer_option_mappings, POINTER_INSENSITIVE);
+
+  config.struct_abstract_type = option_to_abstract_type(
+    options, "structs", struct_option_mappings, STRUCT_INSENSITIVE);
+
+  config.array_abstract_type = option_to_abstract_type(
+    options, "arrays", array_option_mappings, ARRAY_INSENSITIVE);
+
+  config.union_abstract_type = option_to_abstract_type(
+    options, "unions", union_option_mappings, UNION_INSENSITIVE);
+
+  // This should always be on (for efficeny with 3-way merge)
+  // Does not work with value set
+  config.context_tracking.last_write_context =
+    (config.value_abstract_type != VALUE_SET) &&
+    (config.pointer_abstract_type != VALUE_SET);
+  config.context_tracking.data_dependency_context =
+    options.get_bool_option("data-dependencies");
+  config.advanced_sensitivities.new_value_set =
+    options.get_bool_option("new-value-set");
```

Comment:
I know this is a refactor only, but is this option documented in the help output actually?

Reply:
It isn't, because it isn't exposed as a command line option. 

There are unit tests that use this option, but in due course this will go away in favour of additional possible arguments to vsd-values and vsd-pointers.